dotcom-rendering/src/web/components/AdSlot.tsx

```diff
@@ -248,21 +275,23 @@ export const AdSlot = ({
 				case ArticleDisplay.Showcase:
 				case ArticleDisplay.NumberedList: {
 					return (
-						<div
-							id="dfp-ad--right"
-							className={[
-								'js-ad-slot',
-								'ad-slot',
-								'ad-slot--right',
-								'ad-slot--mpu-banner-ad',
-								'ad-slot--rendered',
-								'js-sticky-mpu',
-							].join(' ')}
-							css={adStyles}
-							data-link-name="ad slot right"
-							data-name="right"
-							aria-hidden="true"
-						/>
+						<div css={[adStyles]}>
```

Comment:
Perhaps a silly question, how come the adStyles need to be on this container div as well as the slot?

Reply:
It's to do with the div nesting. On standard articles, dfp-ad--right is nested within the top-right-ad-slot container, so it inherits the styles from there. For inline ads, it inherits from article containers. The ad label didn't show for dfp-ad--right for immersives until I added the div around it and added the styles there. Without adding the styles to the container div the ad just looks like this: <img width="1041" alt="Screenshot 2022-12-01 at 12 16 59" src="https://user-images.githubusercontent.com/108270776/205051420-6cc4561e-319b-4422-91f5-68c413f7893c.png">

Reply:
Ah I see! Thanks :) This may be simplified with another ticket that's been in the backlog for a while to do with having containers on all slots